pkg/observer/observertesthelper/observer_test_helper.go

```diff
@@ -324,13 +324,57 @@ func GetDefaultObserverWithFile(tb testing.TB, ctx context.Context, file, lib st
 	return GetDefaultObserverWithWatchers(tb, ctx, b, opts...)
 }
 
-func GetDefaultSensorsWithFile(tb testing.TB, ctx context.Context, file, lib string, opts ...TestOption) ([]*sensors.Sensor, error) {
+func GetDefaultSensorsWithFile(tb testing.TB, file, lib string, opts ...TestOption) ([]*sensors.Sensor, error) {
 	opts = append(opts, WithConfig(file))
 	opts = append(opts, WithLib(lib))
 
-	b := base.GetInitialSensor()
-	_, sens, err := getDefaultObserverSensors(tb, ctx, b, opts...)
-	return sens, err
+	option.Config.BpfDir = bpf.MapPrefixPath()
+	option.Config.MapDir = bpf.MapPrefixPath()
+
+	testutils.CaptureLog(tb, logger.GetLogger().(*logrus.Logger))
+
+	o := newDefaultTestOptions(opts...)
+
+	option.Config.HubbleLib = os.Getenv("TETRAGON_LIB")
+	if option.Config.HubbleLib == "" {
+		option.Config.HubbleLib = o.observer.lib
+	}
+
+	procfs := os.Getenv("TETRAGON_PROCFS")
+	if procfs != "" {
+		option.Config.ProcFS = procfs
+	}
+
+	if testing.Verbose() {
+		option.Config.Verbosity = 1
+	}
+
+	var tp tracingpolicy.TracingPolicy
+	var err error
+
+	if o.observer.config != "" {
+		tp, err = tracingpolicy.PolicyFromYAMLFilename(o.observer.config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse tracingpolicy: %w", err)
+		}
+	}
+
+	var sensor *sensors.Sensor
+
+	if tp != nil {
+		sensor, err = sensors.GetMergedSensorFromParserPolicy(tp)
```

Comment:
One minor note: I'd like to remove the `GetMergedSensorsFromParserPolicy` and use everywhere the same code path: parse the policy, get a list of sensors, and we load the sensors. No need to change it now, but something to avoid in the fugure.